src/app/util/mock/attribute-storage.cpp

```diff
@@ -0,0 +1,187 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *     This file contains the mock implementation for the generated attribute-storage.cpp
```

Comment:
This is very confusing, because `attribute_storage.cpp` is not generated...  What's generated is the data structure that it traverses.

Note that we have existing macros for building up such a data structure, by the way.  So you could presumably use the existing `attribute-storage.cpp` code directly instead of mocking it...  Followup for considering changing to that.

Reply:
attribute_storage.cpp requires endpoint-config.h, which is generated.

Reply:
Added a comment for this. We should construct a endpoint-config for mock use later.